### PR TITLE
Move `phpstan/phpstan` to `require-dev`

### PR DESCRIPTION
**Pull request by martinssipenko**
## Description

Move `phpstan/phpstan` to `require-dev` so that it's not installed in projects using this package.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read the **[CONTRIBUTING](CONTRIBUTING.md)** document.
- [X] My pull request addresses exactly one patch/feature.
- [X] I have created a branch for this patch/feature.
- [X] Each individual commit in the pull request is meaningful.
- [ ] ~I have added tests to cover my changes.~
- [ ] ~If my change requires a change to the documentation, I have updated it accordingly.~

### CODE REVIEW

**Comment by martinssipenko** [comment]
@Crell I'd really appreciate a release of this change as currently when updating to the latest version of this package, PHPSTan is installed when it should really not be. I caught this because the project where I use this package still uses an older version of PHPStan (0.12.x) and is now conflicting with `1.4.0` of this package.

File: composer.json [hunk 1]
```diff
@@ -54,6 +54,9 @@
         }
     },
     "config": {
-        "sort-packages": true
+        "sort-packages": true,
+        "allow-plugins": {
+            "bamarni/composer-bin-plugin": true
+        }
```
Comment (Crell) [inline composer.json]: Where's this coming from? Is it a new PHPStan dependency or something?

**Comment by Crell** [comment]
Hm, good call.  That was probably just sloppiness on my part, sorry.  Though where is the extra plugin coming from?  Once that's resolved I'm good with merging.

File: composer.json (hunk 1 above)
Comment (martinssipenko) [reply composer.json]: I must have overlooked something, did `compoer update` locally and it went away.

**Comment by martinssipenko** [comment]
CI is failing on 7.4 but I don't believe it's caused by my changes.

**Comment by Crell** [comment]
Kicked CI, works now.  No idea what happened there.

Merged.  Thanks!  I'll tag a new release sometime today; I may as well see if there's any other small cleanup to do while I'm here.

**Pull request merged**